scripts/utils/sign_and_send.js

```diff
@@ -19,6 +19,23 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
     return new Promise((resolve) => rl.question(message, (answer) => resolve(answer)))
   }
 
+  const estimateGas = async function (masterSafe, transaction) {
+    const estimateCall = masterSafe.contract.methods
+      .requiredTxGas(transaction.to, transaction.value, transaction.data, transaction.operation)
+      .encodeABI()
+    const estimateResponse = await web3.eth.call({
+      to: masterSafe.address,
+      from: masterSafe.address,
+      data: estimateCall,
+      gasPrice: 0,
+    })
+    // https://docs.gnosis.io/safe/docs/contracts_tx_execution/#safe-transaction-gas-limit-estimation
+    // The value returned by requiredTxGas is encoded in a revert error message. For retrieving the hex
+    // encoded uint value the first 68 bytes of the error message need to be removed.
+    const txGasEstimate = parseInt(estimateResponse.substring(138), 16)
+    return txGasEstimate
```

Comment:
I would recomment to up this value by at least 1/64 ... due to EIP-150 ;)

Reply:
we experienced some issues when performing gas intensive transactions, therefore we now do a more complex estimation where we double check if the transaction can be performed with the set safeTxGas limit.

Reply:
A little more details to [EIP-150](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-150.md). There is a section that says 

> If a call asks for more gas than the maximum allowed amount (i.e. the total amount of gas remaining in the parent after subtracting the gas cost of the call and memory expansion), do not return an OOG error; instead, if a call asks for more gas than all but one 64th of the maximum allowed amount, call with all but one 64th of the maximum allowed amount of gas (this is equivalent to a version of EIP-901 plus EIP-1142). CREATE only provides all but one 64th of the parent gas to the child call.